src/app.rs

```diff
@@ -576,35 +569,59 @@ impl App {
         }
     }
 
-    pub fn writing_people(&self, channel: &Channel) -> String {
-        if !channel.is_writing() {
-            return String::from("");
-        }
-        let uuids: Vec<Uuid> = match &channel.typing {
-            TypingSet::GroupTyping(hash_set) => hash_set.clone().into_iter().collect(),
-            TypingSet::SingleTyping(a) => {
-                if *a {
-                    vec![channel.user_id().unwrap()]
-                } else {
-                    Vec::new()
+    pub fn writing_people(&self, channel: &Channel) -> Option<String> {
+        if channel.is_writing() {
+            let uuids: Box<dyn Iterator<Item = Uuid>> = match &channel.typing {
+                TypingSet::GroupTyping(uuids) => Box::new(uuids.iter().copied()),
+                TypingSet::SingleTyping(a) => {
+                    if *a {
+                        Box::new(std::iter::once(channel.user_id().unwrap()))
+                    } else {
+                        Box::new(std::iter::empty())
+                    }
                 }
-            }
-        };
-        format!(
-            "{:?} writing...",
-            uuids
-                .into_iter()
-                .map(|u| self.name_by_id(u))
-                .collect::<Vec<&str>>()
-        )
+            };
+            Some(format!(
+                "[{}] writing...",
+                uuids.map(|id| self.name_by_id(id)).format(", ")
+            ))
```

Comment:
Nice UI improvement. We however still have the issue that some "stopped writing" notifications are not received smh. Might open an issue about that.